src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/IProjectSnapshotManager.cs

```diff
@@ -12,12 +14,26 @@ internal interface IProjectSnapshotManager
     event EventHandler<ProjectChangeEventArgs> PriorityChanged;
     event EventHandler<ProjectChangeEventArgs> Changed;
 
+    ImmutableArray<ProjectKey> GetAllProjectKeys(string projectFileName);
```

Comment:
Should we just get rid of these altogether and make everything go through the `ProjectSnapshotManager.Updater` ?

Reply:
No. These are safe to call without going through the dispatcher, thanks your email excellent work!

Reply:
oh yea... duh